### DISCUSSION TITLE
Initial Password for new user

### DISCUSSION THREAD

**Discussion by HamzaMoiyadi** [supabase/supabase#820] [Questions] [answered]
How do I add an initial password for a user using the Email provider? 

Currently I had to switch on signups temporarily and add user emails and password to the auth table.

**Comment by kiwicopple** [comment, marked as answer]
Hey @HamzaMoiyadi the user supplies their own passwords when they sign up: 
https://supabase.io/docs/client/auth-signup#sign-up

```js
const { user, session, error } = await supabase.auth.signUp({
  email: 'example@email.com',
  password: 'example-password',
})
```

You can also send the users a password reset email:

![image](https://user-images.githubusercontent.com/10214025/110230279-d010c700-7f4a-11eb-9216-4ad66026bf03.png)

Does that help? Let us know if you have a use-case where any of these options don't work

**Reply by HamzaMoiyadi** [reply]
Hey, thanks for the quick reply!

I currently have no requirement for sign up on my app, but I need to give folks a temporary password to acess the app. 
By the method above, I guess I'll have to run a script to sign users up instead of using the UI for the same 😅
